snap_joint.scad

```diff
@@ -0,0 +1,58 @@
+include <constants.scad>
+use <util.scad>
+
+module cantilever(length, width, thickness, overhang) {
+  inner_length = length - overhang;
+  linear_extrude(width) {
+    polygon([
+      [0, 0],
+      [inner_length, 0],
+      [inner_length, -overhang],
+      [length, 0],
+      [length, thickness],
+      [0, thickness],
+    ]);
+    for (i=[0:1])
+      translate([0, (i*2 - 0.5)*thickness, 0])
```

Comment:
Two things: one, you can probably use mirrored([y]) here, though it'd still have to have a translate since the cantilever is aligned on y and these points aren't.

Two, you might want to make this a module, like "module curved_armpit" :) seems like it could be useful elsewhere!